src/main/java/htsjdk/samtools/cram/build/CompressionHeaderFactory.java

```diff
@@ -479,56 +479,56 @@ private EncodingDetails buildEncodingForTag(final List<CramCompressionRecord> re
     private static class CompressionHeaderBuilder {
         private final CompressionHeader header;
 
-        CompressionHeaderBuilder(final boolean sorted) {
+        CompressionHeaderBuilder(final boolean coordinateSorted) {
             header = new CompressionHeader();
             header.externalIds = new ArrayList<>();
             header.tMap = new TreeMap<>();
 
             header.encodingMap = new TreeMap<>();
-            header.APDelta = sorted;
+            header.APDelta = coordinateSorted;
         }
 
         CompressionHeader getHeader() {
             return header;
         }
 
-        void addExternalEncoding(final EncodingKey encodingKey, final EncodingParams params,
+        void addExternalEncoding(final DataSeries dataSeries, final EncodingParams params,
                                  final ExternalCompressor compressor) {
-            header.externalIds.add(encodingKey.ordinal());
-            header.externalCompressors.put(encodingKey.ordinal(), compressor);
-            header.encodingMap.put(encodingKey, params);
+            header.externalIds.add(dataSeries.ordinal());
```

Comment:
Agreed - there's so specific int values in the spec that we need to match.

Unfortunately after looking at this, it's not a simple change.  These fields can also be integers which refer to `ReadTag`s instead of `DataSeries` and I don't yet understand how that all interacts.